src/wallet/rpc/wallet.cpp

```diff
@@ -727,16 +729,28 @@ static RPCHelpMan migratewallet()
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
         {
-            std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
-            if (!wallet) return NullUniValue;
-
-            if (wallet->IsCrypted()) {
-                throw JSONRPCError(RPC_WALLET_WRONG_ENC_STATE, "Error: migratewallet on encrypted wallets is currently unsupported.");
+            std::string wallet_name;
+            if (GetWalletNameFromJSONRPCRequest(request, wallet_name)) {
+                if (!(request.params[0].isNull() || request.params[0].get_str() == wallet_name)) {
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "RPC endpoint wallet and wallet_name parameter specify different wallets");
+                }
+            } else {
+                wallet_name = request.params[0].get_str();
             }
 
             WalletContext& context = EnsureWalletContext(request.context);
+            std::shared_ptr<CWallet> wallet = GetWallet(context, wallet_name);
+            util::Result<MigrationResult> res;
+            if (wallet) {
+                if (wallet->IsCrypted()) {
+                    throw JSONRPCError(RPC_WALLET_WRONG_ENC_STATE, "Error: migratewallet on encrypted wallets is currently unsupported.");
+                }
+
+                res = MigrateLegacyToDescriptor(std::move(wallet), context);
+            } else {
+                res = MigrateLegacyToDescriptor(wallet_name, context);
+            }
```

Comment:
In 433f9f8d7539efc26b952063d6127315606d248e "rpc: Allow users to specify wallet name for migratewallet"

These lines don't seem to have any test coverage. Also, if `GetWallet` is unable to get the wallet, shouldn't an error be thrown at that point?

Reply:
> These lines don't seem to have any test coverage.

Tests are added in a later commit.

> Also, if `GetWallet` is unable to get the wallet, shouldn't an error be thrown at that point?

No, `GetWallet` only looks for wallets that have been loaded. But it is necessary to be able to migrate wallets that are not currently loaded. `MigrateLegacyToDescriptor` will try loading the wallet and result in an error if it does not actually exist.